modules/lidar_oscillation.py

```diff
@@ -0,0 +1,50 @@
+"""
+Representing a LiDAR Oscillation
+"""
+
+from lidar_detection import LidarDetection
+
+
+class LidarOscillation:
+    """
+    Class to represent a collection of LiDAR readings that make up an oscillation.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(cls, readings: list[LidarDetection]) -> "tuple[bool, LidarOscillation | None]":
+        """
+        Create a new LidarOscillation object from a list of LidarReading objects.
+        """
+        # Ensuring the list does not contain None values
+        if any(reading is None for reading in readings):
```

Comment:
we can assume that every reading inside readings is valid so no need to check here. validation will be done in the lidar parser, not here in the data structure creation itself